src/evaluate.py

```diff
@@ -0,0 +1,184 @@
+        model_output (str or dict): The YAML configuration generated by the model.
+
+    Returns:
+        int: The Levenshtein distance between the two YAML configurations.
+    """
+    # Convert dictionaries to YAML strings if they are not already strings
+    if isinstance(expected_answer, dict):
+        expected_answer = yaml.dump(expected_answer)
+    if isinstance(model_output, dict):
+        model_output = yaml.dump(model_output)
+
+    # Calculate and return the Levenshtein distance
+    return Levenshtein.distance(expected_answer, model_output)
+
+
+def evaluate_model(evaluation_tokenizer, trained_model, processed_data, metric_types):
+    question_list = []
+    expected_answer_list = []
+    generated_answer_list = []
+    data_len = len(processed_data)
+    for data in processed_data:
+        question = data["question"]
+        question_list.append(question)
+        expected_answer = data["answer"]
+        expected_answer_list.append(expected_answer)
+        answer = generate_answer(question, evaluation_tokenizer, trained_model)
+        generated_answer_list.append(answer)
+
+    results = {}
+    for metric_type in metric_types:
+        if metric_type == "EM":
+            results["EM"] = (
+                sum(
+                    exact_match_score(et, pt)
+                    for et, pt in zip(expected_answer_list, generated_answer_list)
+                )
+                / data_len
+            )
+        elif metric_type == "Schema Correct":
+            results["Schema_Correct"] = (
+                sum(schema_correct(pt) for pt in generated_answer_list) / data_len
+            )
+        elif metric_type == "Levenshtein Distance":
+            results["Levenshtein_Distance"] = (
+                sum(
+                    levenshtein_distance(et, pt)
+                    for et, pt in zip(expected_answer_list, generated_answer_list)
+                )
+                / data_len
+            )
+
+    return results
+
+
+def generate_answer(question, evaluation_tokenizer, trained_model):
+    # Tokenize the question
+    inputs = evaluation_tokenizer.encode(question, return_tensors="pt")
+
+    # Generate the answer
+    outputs = trained_model.generate(inputs, max_length=500)
+    answer = evaluation_tokenizer.decode(outputs[0], skip_special_tokens=True)
+
+    # Removing the repeated question if it appears in the answer
+    cleaned_answer = answer.replace(question, "").strip()
+
+    return cleaned_answer
+
+
+def write_results_to_csv(results, filename):
+    # Open the file in write mode
+    with open(filename, mode="w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        # Write the header
+        writer.writerow(["Metric", "Score"])
+        # Iterate over the dictionary and write rows
+        for key, value in results.items():
+            writer.writerow([key, value])
+
+
+def main():
+    # Load tokenizer and model
+    model_path = "../model/evaluation"
```

Comment:
These should be arguments to the script to ensure they can be easily adjusted.

Reply:
Sure, btw, yesterday, I failed to upload the model to the github, it seems to me that I need to use `git lfs` to store the model. And in the  github action integration testing, we also need to use `git lfs` to pull the model to the  host machine to test it.

Reply:
How big is the model?